### PR TITLE
Added program schedules and handling of program reruns

### PR DESCRIPTION
**Pull request by madsthom**
Added program schedules and handling of program reruns

### CODE REVIEW

**Comment by clhols** [comment]
Please try to use DrMuReactiveRepository instead of DrMuRepository.

**Comment by madsthom** [comment]
Also changed the getScheduleObservable to take the date parameter as String instead of Date.

**Comment by clhols** [comment]
Instead of calling api.getScheduleObservable(id, SimpleDateFormat("yyyy-MM-dd HH:MM:ss", Locale.GERMAN).format(Date())) as you do now, you could have kept the Date type and called api.getScheduleObservable(id, Date()). Then the knowledge of how the date string should be formatted is avoided in the API.

File: app/src/main/java/dk/youtec/drchannels/backend/DrMuReactiveRepository.kt [hunk 1]
```diff
@@ -67,7 +67,6 @@ class DrMuReactiveRepository(context: Context) {
             try {
                 val dateString = SimpleDateFormat("yyyy-MM-dd HH:MM:ss", Locale.GERMAN)
                         .format(date)
-
```
Comment (clhols) [inline app/src/main/java/dk/youtec/drchannels/backend/DrMuReactiveRepository.kt]: Avoid changes that isn't required to meet the goal

File: app/src/main/java/dk/youtec/drchannels/ui/MainActivity.kt [hunk 2]
```diff
@@ -46,7 +46,6 @@ class MainActivity : AppCompatActivity(), AnkoLogger, ChannelsAdapter.OnChannelC
 
     private lateinit var viewModel: ChannelsViewModel
     private var adapter: ChannelsAdapter? = null
-
```
Comment (clhols) [inline app/src/main/java/dk/youtec/drchannels/ui/MainActivity.kt]: Avoid changes that isn't required to meet the goal

File: app/src/main/java/dk/youtec/drchannels/ui/MainActivity.kt [hunk 3]
```diff
@@ -152,14 +151,12 @@ class MainActivity : AppCompatActivity(), AnkoLogger, ChannelsAdapter.OnChannelC
         // Inflate the menu; this adds items to the action bar if it is present.
         menuInflater.inflate(R.menu.menu_main, menu)
 
-        /*
```
Comment (clhols) [inline app/src/main/java/dk/youtec/drchannels/ui/MainActivity.kt]: Avoid changes that isn't required to meet the goal

File: app/src/main/java/dk/youtec/drchannels/ui/MainActivity.kt [hunk 4]
```diff
@@ -152,14 +151,12 @@ class MainActivity : AppCompatActivity(), AnkoLogger, ChannelsAdapter.OnChannelC
...
-        val listsMenu = menu.findItem(R.id.menu_favourite_lists)
+        /*val listsMenu = menu.findItem(R.id.menu_favourite_lists)
         val subMenu = listsMenu.subMenu
         for ((key, name) in favouriteLists) {
             subMenu.add(listsGroup, key, Menu.NONE, name)
         }
-        listsMenu.isVisible = favouriteLists.isNotEmpty()
-        */
```
Comment (clhols) [inline app/src/main/java/dk/youtec/drchannels/ui/MainActivity.kt]: Avoid changes that isn't required to meet the goal

File: app/src/main/java/dk/youtec/drchannels/ui/MainActivity.kt [hunk 5]
```diff
@@ -246,8 +243,9 @@ class MainActivity : AppCompatActivity(), AnkoLogger, ChannelsAdapter.OnChannelC
     }
 
     override fun showChannel(context: Context, channel: MuNowNext) {
-        val intent = Intent(context, ProgramsActivity::class.java).apply {
-            putExtra(ProgramsActivity.CHANNEL_NAME, channel.Channel)
+        val intent = Intent(context, ProgramsActivity::class.java)
```
Comment (clhols) [inline app/src/main/java/dk/youtec/drchannels/ui/MainActivity.kt]: Don't change apply() to with() call

File: app/src/main/java/dk/youtec/drchannels/ui/ProgramsActivity.kt [hunk 6]
```diff
@@ -49,27 +52,38 @@ class ProgramsActivity : AppCompatActivity() {
...
         doAsync {
             val id = intent.extras.get(CHANNEL_ID) as String
-            val programs = api.getSchedule(id, SimpleDateFormat("yyyy-MM-dd HH:MM:ss", Locale.GERMAN).format(Date()))
+            disposables.add(
+                    api.getScheduleObservable(id, Date())
+                            .subscribeOn(Schedulers.io())
+                            .observeOn(AndroidSchedulers.mainThread())
+                            .subscribeBy(
+                                    onNext = { programs ->
+                                        uiThread {
```
Comment (clhols) [inline app/src/main/java/dk/youtec/drchannels/ui/ProgramsActivity.kt]: uiThread isn't needed as you observe on mainThread

File: app/src/main/java/dk/youtec/drchannels/ui/adapter/ChannelsAdapter.kt [hunk 7]
```diff
@@ -31,11 +31,7 @@ import java.util.*
 fun ViewGroup.inflate(@LayoutRes layoutRes: Int, attachToRoot: Boolean = false): View =
         LayoutInflater.from(context).inflate(layoutRes, this, attachToRoot)
 
-class ChannelsAdapter(
-        val contentView: View?,
-        var channels: List<MuNowNext>,
-        val listener: OnChannelClickListener
-) : RecyclerView.Adapter<ChannelsAdapter.ViewHolder>() {
+class ChannelsAdapter(val contentView: View?, var channels: List<MuNowNext>, val listener: OnChannelClickListener) : RecyclerView.Adapter<ChannelsAdapter.ViewHolder>() {
```
Comment (clhols) [inline app/src/main/java/dk/youtec/drchannels/ui/adapter/ChannelsAdapter.kt]: Avoid changes that isn't required to meet the goal

File: app/src/main/java/dk/youtec/drchannels/ui/adapter/ProgramAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+import dk.youtec.drchannels.backend.DrMuReactiveRepository
+import io.reactivex.disposables.CompositeDisposable
+import dk.youtec.drapi.*
+
+
+class ProgramAdapter(val context: Context, val channelSID: String, val programs: Schedule, val api: DrMuReactiveRepository) : RecyclerView.Adapter<ProgramAdapter.ViewHolder>() {
+
+    private var mColorMatrixColorFilter: ColorMatrixColorFilter
+    private var mResources: Resources
+    private val disposables = CompositeDisposable()
```
Comment (clhols) [inline app/src/main/java/dk/youtec/drchannels/ui/adapter/ProgramAdapter.kt]: disposables isn't used as you don't have an onDestory to clear() is with. Like MainActivity has.

**Pull request merged**